geode-redis/src/test/java/org/apache/geode/redis/internal/data/RedisStringTest.java

```diff
@@ -41,16 +42,228 @@ public static void beforeClass() {
+    RedisString string = new RedisString(byteArrayWrapper);
+    assertThatThrownBy(() -> string.incrby(region, byteArrayWrapper, 2L))
+        .isInstanceOf(NumberFormatException.class);
+  }
+
+  @Test
+  public void incrbyErrorsWhenValueOverflows() {
+    // allows unchecked cast of mock to Region<ByteArrayWrapper, RedisData>
+    @SuppressWarnings("unchecked")
+    Region<ByteArrayWrapper, RedisData> region = mock(Region.class);
+    ByteArrayWrapper byteArrayWrapper = new ByteArrayWrapper(
+        // max value for signed long
+        new byte[] {'9', '2', '2', '3', '3', '7', '2', '0', '3', '6', '8', '5', '4', '7', '7', '5',
+            '8', '0', '7'});
+    RedisString string = new RedisString(byteArrayWrapper);
+    assertThatThrownBy(() -> string.incrby(region, byteArrayWrapper, 2L))
+        .isInstanceOf(ArithmeticException.class);
+  }
+
+  @Test
+  public void incrbyIncrementsValueByGivenLong() {
+    // allows unchecked cast of mock to Region<ByteArrayWrapper, RedisData>
+    @SuppressWarnings("unchecked")
+    Region<ByteArrayWrapper, RedisData> region = mock(Region.class);
+    ByteArrayWrapper byteArrayWrapper = new ByteArrayWrapper(new byte[] {'1', '0'});
+    RedisString string = new RedisString(byteArrayWrapper);
+    string.incrby(region, byteArrayWrapper, 2L);
+    assertThat(string.get().toString()).isEqualTo("12");
+  }
+
+  @Test
+  public void incrbyfloatThrowsArithmeticErrorWhenNotADouble() {
+    // allows unchecked cast of mock to Region<ByteArrayWrapper, RedisData>
+    @SuppressWarnings("unchecked")
+    Region<ByteArrayWrapper, RedisData> region = mock(Region.class);
+    ByteArrayWrapper byteArrayWrapper = new ByteArrayWrapper(new byte[] {'1', '0', ' ', '1'});
+    RedisString string = new RedisString(byteArrayWrapper);
+    assertThatThrownBy(() -> string.incrbyfloat(region, byteArrayWrapper, 1.1))
+        .isInstanceOf(NumberFormatException.class);
+  }
+
+  @Test
+  public void incrbyfloatIncrementsValueByGivenFloat() {
+    // allows unchecked cast of mock to Region<ByteArrayWrapper, RedisData>
+    @SuppressWarnings("unchecked")
+    Region<ByteArrayWrapper, RedisData> region = mock(Region.class);
+    ByteArrayWrapper byteArrayWrapper = new ByteArrayWrapper(new byte[] {'1', '0'});
+    RedisString string = new RedisString(byteArrayWrapper);
+    string.incrbyfloat(region, byteArrayWrapper, 2.20);
+    assertThat(string.get().toString()).isEqualTo("12.2");
+  }
+
+  @Test
+  public void decrErrorsWhenOverflows() {
+    // allows unchecked cast of mock to Region<ByteArrayWrapper, RedisData>
+    @SuppressWarnings("unchecked")
+    Region<ByteArrayWrapper, RedisData> region = mock(Region.class);
+    ByteArrayWrapper byteArrayWrapper = new ByteArrayWrapper(new byte[] {0});
+    RedisString string = new RedisString(byteArrayWrapper);
+    assertThatThrownBy(() -> string.decr(region, byteArrayWrapper))
+        .isInstanceOf(NumberFormatException.class);
```

Comment:
This seems to be an error due to the inability to parse the value as a long rather than due to overflow.  We could change the test name or make it error due to overflow instead.